iterators task (1)/student/include/impl.h

```diff
@@ -2,20 +2,77 @@
 #include <iterator>
 
 template <class Container>
-struct accumulator
+struct accumulator 
 {
-    /** @todo Iterator traits*/
-    /** @todo Constructor with container*/
-    /** @todo Copy assignment operator*/
-    /** @todo Move assignment operator*/
+private:
+    Container& mContainer;
+public:
+    /** Iterator traits*/
+    using iterator_category = std::output_iterator_tag;
+    using value_type =  typename Container::value_type;
```

Comment:
void is usually used for output iterators here
will your solution work with c arrays?

Reply:
Changed to void.

->will your solution work with c arrays?
Even after change it will not work with c arrays as they do not have neither insert nor push_back methods. Container there  is a type of object where we put data.

Reply:
agree